### PR TITLE
Adding Periodic Run and Issue Creation for Clippy

### PR DESCRIPTION
**Pull request by peytonr18**
Adding weekly scheduled run mostly for the purposes of running clippy periodically.
Also amending the workflow so that if clippy discovers a warning in the codebase, it will create an issue for that run.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -34,8 +36,12 @@ jobs:
...
-          args: --verbose -- --deny warnings
+      - name: Run Clippy and redirect output to file
+        run: cargo clippy --verbose -- --deny warnings > clippy.log 2>&1 || true
+      - name: Create issue if clippy found warnings
+        run: |
+          if [ $(grep -c 'warning' clippy.log || true) != '0' ]; then
+            gh issue create --title "Clippy discovered warnings in the codebase" --body-file clippy.log
+          fi
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (dongsupark) [inline .github/workflows/ci.yaml:47]: It means, this GitHub repo must have a personal access token `GITHUB_TOKEN` configured for that. I do not have that privilege.

**Review by dongsupark — COMMENTED** [review]
Thanks. The idea of periodic running look ok in general.

However I am wondering exactly what you want to achieve.
Do you want to run `cargo clippy` not when creating a PR, and only periodically?
Or do you want to run it both on PR creation and periodically?

In case of former, you would probably like to split the periodic part into a separate workflow.
In case of latter, running `cargo clippy` twice sounds a little superfluous to me.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (peytonr18) [reply .github/workflows/ci.yaml:47]: Sorry for not updating this sooner - I was communicating with my team and I think we are going to back off the plan to create an issue for clippy warnings. There is a org-wide movement to move away from using tokens/secrets of any kind and introducing another one right now is probably not the best idea.

Comment (peytonr18) [reply .github/workflows/ci.yaml:47]: In terms of running clippy periodically, we can split it into a separate workflow if you think that would be best. I was planning on just having the entire CI workflow run every Sunday at midnight in addition to on PR creation / pushes to main just to minimize the number of workflows we have running. But, if we think it's best to just split it up, it's not hard to do and I can close this PR and just create a separate clippy workflow.

**Pull request closed without merging**